src/content/docs/dev/general-message-passing/sui/gmp-tutorial.mdx

```diff
@@ -0,0 +1,918 @@
+console.log('📦 Published package ID:', publishedChange?.packageId);
+
+
+const [gmpSingletonObjectId] = getObjectIdsByObjectTypes(response, [
+`${publishedChange.packageId}::gmp::Singleton`,
+]);
+console.log('🔑 GMP Singleton Object ID:', gmpSingletonObjectId);
+```
+
+The `packageId` can be found by going through the `objectChanges` in the `response` for the signed transaction. The `gmpSinlgetonObjectId` however requires a helper function called `getObjectIdsByObjectTypes()`.
+
+This helper function that is yet to be defined will find a specified object in a given transaction and return the object's `objectId`. In your case you need to fine the `objectId` for the `singleton`. The function can be written out as follows:
+
+```js
+export const getObjectIdsByObjectTypes = (txn, objectTypes) =>
+  objectTypes.map((objectType) => {
+    const objectId = txn.objectChanges.find((change) =>
+      change.objectType?.includes(objectType)
+    )?.objectId;
+
+    if (!objectId) {
+      throw new Error(`No object found for type: ${objectType}`);
+    }
+    return objectId;
+})
+```
+
+
+Great! At this point when you run `node scripts/deploy.js` you should see the following logs.
+
+```bash
+node scripts/deploy.js
+📦 Building Move package
+INCLUDING DEPENDENCY GasService
+INCLUDING DEPENDENCY RelayerDiscovery
+INCLUDING DEPENDENCY AxelarGateway
+INCLUDING DEPENDENCY Utils
+INCLUDING DEPENDENCY VersionControl
+INCLUDING DEPENDENCY Sui
+INCLUDING DEPENDENCY MoveStdlib
+BUILDING gmp_example
+🚀 Sending publish transaction…
+✅ Publish succeeded!
+📦 Published package ID: 0x9068ae5fb08615d7fd9f869d6d1cf47ec21fa687ca7ab3d7d68b45a7b2cfd041
+🔑 GMP Singleton Object ID: 0x9cbb3582c80ed49ca5d0cacce23b2cdb145bd10421f2c969d36799e3cbd78c52
+```
+
+
+## Interact With Contract
+
+The final (and most exciting) thing that needs to be done is to actually interact with the package now that it's live on-chain. In your `scripts` directory, next to your `deploy.js` script you can create a new script called `gmp.js`.
+
+### Setup
+
+This script will also user the `commander` tool to execute
+
+#### Command Setup
+
+
+
+Similarly to the `deploy` script, you can begin by setting up a new `Command`. This command however will require significantly more parameters than the deployment did, so as to specify the params for the `send_message()` function in your `move` package.
+
+This will prompt you to pass in 
+1. The `destination chain`
+1. The `destination address` on the destination chain
+1. The `id` of the singleton object (that should be logged in your deployment script)
+1. The `fee` you are paying to the [axelar gas service](/dev/general-message-passing/sui/gmp-contracts/#sui-gas-service) when sending the cross-chain transaction.
+1. The `payload` for the gmp message being sent
+1. The `refundAddress` to receive any surplus funds sent to the gas service 
+1. The `packageId` of your live package on the Sui blockchain
+
+```js
+const program = new Command()
```

Comment:
Where should they add this? How did we get here, sorry?I  just realised we didn't clone any project or start a project from scratch. Am i missing something?

Reply:
in deploy.js